### ISSUE TITLE
Functional extension: guide the process by a 'pattern matching'-like construct

### ISSUE THREAD

**Issue by gvandeglind** [jgordijn/process#6]
e.g. http://docs.scala-lang.org/tutorials/tour/pattern-matching.html

**Comment by jgordijn** [comment]
Yes, that would be nice. Maybe we can first write a test case for it to see when in particular this will be useful.

**Comment by gvandeglind** [comment]
Yes, that's a good idea. I think once there is support for pattern matching I would (often) prefer this over the if statement, since it will make the process more readable particularly when there are multiple (more then 2) cases (which would lead to nested if constructs).

**Comment by gvandeglind** [comment]
Issue is in progress...I would like to put a label on it that I am working on it (it's in progress), but probably I don't have the right to do so ;-)

**Comment by jgordijn** [comment]
I added you as collaborator @gvandeglind . Can you try again?

**Comment by gvandeglind** [comment]
Thanks! I have assigned it to myself. btw, the pattern matching is almost working...I still need to extend the DSL to be able to actually use the new construct.
